dali/pipeline/workspace/device_workspace.cc

```diff
@@ -36,6 +36,24 @@ const TensorList<GPUBackend>& DeviceWorkspace::Input(int idx) const {
   return *gpu_inputs_[tensor_meta.second];
 }
 
+template <>
+TensorList<CPUBackend>& DeviceWorkspace::MutableInput(int idx) {
```

Comment:
Don't like the idea of MutableInputs accessible from operators, as this breaks some contracts about our execution.

Reply:
I agree, the sole existence of the function is hacky and not elegant.
However, its usage doesn't break the contract , since `Input` are guaranteed to not change in the Run scope. That means data and metadata should be unchanged at the end of `Run`, which is the case.
What I propose is keeping it until we have a full integrated support of the Op/Kernel logic.